quickwit-search/src/collector.rs

```diff
@@ -280,16 +280,18 @@ impl Collector for QuickwitCollector {
         // starting from 0 for every leaves.
         let leaf_max_hits = self.max_hits + self.start_offset;
 
-        let timestamp_filter_opt = if let Some(timestamp_field) = self.timestamp_field_opt {
-            TimestampFilter::new(
+        // Create a timestamp filter only if necessary as, once created,
+        // we will read some bytes in the fast field file that are not in the hotcache.
+        let timestamp_filter_opt = self.timestamp_field_opt
+            .filter(|_| self.start_timestamp_opt.is_some() || self.end_timestamp_opt.is_some())
+            .map(|timestamp_field| TimestampFilter::new(
                 timestamp_field,
                 self.start_timestamp_opt,
                 self.end_timestamp_opt,
                 segment_reader,
-            )?
-        } else {
-            None
-        };
+            ))
+            .transpose()?
```

Comment:
The `transpose()?.flatten()` is way less readable.

Reply:
Yes it's horrible. I finally came back to 2 good old `if else` conditions, I generally tried to use as few as possible nested if conditions, but in this case, I think this is ok.